tests/network/primary_pod_network.go

```diff
@@ -0,0 +1,150 @@
+
+	BeforeEach(func() {
+		var err error
+		virtClient, err = kubecli.GetKubevirtClient()
+		Expect(err).NotTo(HaveOccurred(), "Should successfully initialize an API client")
+	})
+
+	Describe("Status", func() {
+		AssertReportedIP := func(vmi *v1.VirtualMachineInstance) {
+			By("Getting pod of the VMI")
+			vmiPod := tests.GetRunningPodByVirtualMachineInstance(vmi, tests.NamespaceTestDefault)
+
+			By("Making sure IP reported on the VMI matches the one on the pod")
+			Expect(vmi.Status.Interfaces[0].IP).To(Equal(vmiPod.Status.PodIP))
+			Expect(vmi.Status.Interfaces[0].IPs).NotTo(BeEmpty())
+			Expect(vmi.Status.Interfaces[0].IPs[0]).To(Equal(vmiPod.Status.PodIP))
+		}
+
+		Context("VMI connected to the pod network using the default (implicit) binding", func() {
+			var vmi *v1.VirtualMachineInstance
+
+			BeforeEach(func() {
+				vmi = setupVMI(virtClient, vmiWithDefaultBinding())
+			})
+
+			AfterEach(func() {
+				cleanupVMI(virtClient, vmi)
+			})
+
+			It("should report PodIP as its own on interface status", func() { AssertReportedIP(vmi) })
+		})
+
+		Context("VMI connected to the pod network using bridge binding", func() {
+			var vmi *v1.VirtualMachineInstance
+
+			BeforeEach(func() {
+				vmi = setupVMI(virtClient, vmiWithBridgeBinding())
+			})
+
+			AfterEach(func() {
+				cleanupVMI(virtClient, vmi)
+			})
+
+			It("should report PodIP as its own on interface status", func() { AssertReportedIP(vmi) })
+		})
+
+		Context("VMI connected to the pod network using masquerade binding", func() {
+			var vmi *v1.VirtualMachineInstance
+
+			BeforeEach(func() {
+				vmi = setupVMI(virtClient, vmiWithMasqueradeBinding())
+			})
+
+			AfterEach(func() {
+				cleanupVMI(virtClient, vmi)
+			})
+
+			It("should report PodIP as its own on interface status", func() { AssertReportedIP(vmi) })
+		})
+	})
+})
+
+func setupVMI(virtClient kubecli.KubevirtClient, vmi *v1.VirtualMachineInstance) *v1.VirtualMachineInstance {
+	By("Creating the VMI")
+	var err error
+	vmi, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(vmi)
+	Expect(err).NotTo(HaveOccurred(), "VMI should be successfully created")
+
+	By("Waiting until the VMI gets ready")
+	vmi = tests.WaitUntilVMIReady(vmi, tests.LoggedInAlpineExpecter)
+
+	return vmi
+}
+
+func cleanupVMI(virtClient kubecli.KubevirtClient, vmi *v1.VirtualMachineInstance) {
+	if vmi != nil {
+		By("Deleting the VMI")
+		Expect(virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Delete(vmi.GetName(), &metav1.DeleteOptions{})).To(Succeed())
+
+		By("Waiting for the VMI to be gone")
+		Eventually(func() bool {
+			_, err := virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Get(vmi.GetName(), &metav1.GetOptions{})
+			if errors.IsNotFound(err) {
+				return true
+			}
+			Expect(err).ToNot(HaveOccurred(), "Should be able to get the VMI from API server")
+			return false
+		}, 6*time.Minute, time.Second).Should(BeTrue(), "The VMI should be gone within the given timeout")
```

Comment:
I think you are looking for `WaitForVirtualMachineToDisappearWithTimeout`, right?

If you are looking to re-implement it here in order to improve it, then I would suggest just returning the error and using `WithTransform`.

`Should(WithTransform(errors.IsNotFound, BeTrue())`

What do you think?

Reply:
Oh I was looking for something like that, thanks!